app/src/main/java/com/github/swent/echo/viewmodels/event/EventViewModel.kt

```diff
@@ -29,30 +43,77 @@ class EventViewModel() : ViewModel() {
+                Log.e("create event", "the user is not logged in")
+            } else {
+                _event.value =
+                    _event.value.copy(
+                        creatorId = userid,
+                        organizerId = userid,
+                        organizerName = repository.getUserProfile(userid).name
+                    )
+                _status.value = EventStatus.New
+            }
+            allTagsList.value = repository.getAllTags()
+        }
+    }
 
-    // return the event
-    fun getEvent(): Event {
-        return _event.asStateFlow().value
+    // constructor for an already existing event
+    constructor(
+        repository: Repository,
+        authenticationService: AuthenticationService,
+        eventId: String
+    ) : this(repository, authenticationService) {
+        _status.value = EventStatus.Saving // avoid saving before initialization
+        viewModelScope.launch {
+            _event.value = repository.getEvent(eventId)
+            _status.value = EventStatus.Saved
+        }
     }
 
-    // return the organizer name of the event
-    fun getOrganizerName(): String {
-        // TODO: get organizer name from repository
-        return ""
+    // return the event
+    fun getEvent(): StateFlow<Event> {
+        return _event.asStateFlow()
     }
 
     // return the list of possible organizer for the user
-    fun getOrganizerList(): List<String> {
-        // placeholder until we get organizer list from repository
-        return listOf("testOrganizer", "anotherTestOrganizer")
+    fun getOrganizerList(): StateFlow<List<String>> {
+        // TODO: also return the associations linked to the user (not implemented yet in the
+        // repository)
+        val organizerList = MutableStateFlow(listOf(_event.value.organizerName))
+        return organizerList
     }
 
     // set the organizer of the event
-    fun setOrganizer(organizer: String) {
-        // TODO: set organizer of the event (need repository)
+    fun setOrganizer(organizerName: String) {
+        viewModelScope.launch {
+            var organizerId = ""
+            val creatorName = repository.getUserProfile(_event.value.creatorId).name
+            if (creatorName == organizerName) {
+                organizerId = _event.value.creatorId
+            } else {
+                val association =
+                    repository.getAllAssociations().find { a -> a.name == organizerName }
+                if (association != null) {
+                    organizerId = association.associationId
+                }
+            }
+            if (organizerId != "") {
+                setEvent(
+                    _event.value.copy(organizerId = organizerId, organizerName = organizerName)
+                )
+            } else {
+                Log.e("set event organizer", "organizer not found in the repository")
+            }
+        }
```

Comment:
Do we need to do all those checks as the user can only pick from the organizers list you show to him. If we only show him the organizations he's in, he won't be able to chose an incompatible association. In the worst case (he gets kicked out of the association during the creation phase) the security rules of supabase will prevent saving to the database.

Maybe this function is not even necessary as the composable will simply call `setEvent` with the organizer field already set according to the users actions.

Reply:
Thanks for the reasoning, I changed this function to set only the creator as organizer and added a todo until we get #84 merged to add association as organizer